### PR TITLE
[BEAM] Making sure that GcsBufferedReader implements the iterator protocol

### PR DESCRIPTION
**Pull request by pabloem**
The change adds only the `__iter__` and `next` functions to the `GcsBufferedReader` class, so that it implements the iterator protocol in Python.

### CODE REVIEW

**Comment by pabloem** [comment]
@chamikaramj This is ready for review : )

File: sdks/python/apache_beam/io/gcsio.py [hunk 1]
```diff
@@ -307,6 +307,14 @@ def __init__(self,
   def _get_object_metadata(self, get_request):
     return self.client.objects.Get(get_request)
 
+  def __iter__(self):
+    return self
+
+  def next(self):
+    """Read one line delimited by '\\n' from the file.
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/gcsio.py]: '\n' instead of '\ \ n'

File: sdks/python/apache_beam/io/gcsio.py [hunk 2]
```diff
@@ -307,6 +307,14 @@ def __init__(self,
...
+    """
+    return self.readline()
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/gcsio.py]: I believe you have to raise 'StopIteration' at the EOF.

File: sdks/python/apache_beam/io/gcsio.py (hunk 1 above)
Comment (pabloem) [inline sdks/python/apache_beam/io/gcsio.py]: I saw `\\n` in the docs for readline. Should I change it in both places?

File: sdks/python/apache_beam/io/gcsio_test.py [hunk 3]
```diff
@@ -341,6 +341,36 @@ def test_file_random_seek(self):
...
+
+    contents = ''.join(lines)
+    bucket, name = gcsio.parse_gcs_path(file_name)
+    self.client.objects.add_file(FakeFile(bucket, name, contents, 1))
+
+    f = self.gcs.open(file_name)
+
+    stop_it_thrown = False
+    read_lines = 0
+    while True:
```
Comment (pabloem) [inline sdks/python/apache_beam/io/gcsio_test.py]: After offline discussion, agreed on using `for` loop. Done.

File: sdks/python/apache_beam/io/gcsio.py [hunk 4]
```diff
@@ -307,6 +307,22 @@ def __init__(self,
...
+    """Read one line delimited by '\\n' from the file.
+    """
+    return self.next()
+
+  def next(self):
+    """Read one line delimited by '\\n' from the file.
+    """
+    line = self.readline()
+    if not line:
+      raise StopIteration()
```
Comment (pabloem) [inline sdks/python/apache_beam/io/gcsio.py]: Done.

**Comment by chamikaramj** [comment]
LGTM. 

Looks like Travis failure is unrelated.

**Comment by pabloem** [comment]
@robertwb should we merge this? Or should we wait to fix the `tox` issue on Mac?

**Comment by robertwb** [comment]
Can you bump it with an empty commit and see if tox is still an issue?

**Comment by pabloem** [comment]
I squashed all the commits together. The tox issue still reproduces. Note that tox runs on Linux and flags no issues whatsoever. : )

**Comment by pabloem** [comment]
I have also a [code change](https://github.com/apache/incubator-beam/pull/1068) that fixes the `tox` issue.

**Comment by pabloem** [comment]
Relaunching tests.

**Pull request closed without merging**